internal/observability/event/options.go

```diff
@@ -70,6 +71,13 @@ func WithFlush() Option {
 	}
 }
 
+// WithInfo allows an optional map as info
```

Comment:
I'm happy to add a bit more docs... but they are only used in WriteObservation(...) where there usage is fairly explicit.  If it's not explict there then I need to add more docs there def.

Reply:
Just based on the names, "Info" and "Detail" feel a bit synonymous. I see that the doc for `WriteObersvation` states:

> At least one and any combination of the supported options may be used:
> WithHeader, WithDetails, WithId, WithFlush and WithRequestInfo. All other
> options are ignored.

Which means for `WriteObservation` `WithDetails` should be used and not `WithInfo`. 

The doc for `WriteError` should probably be updated since it currently says:

> The options WithId and WithRequestInfo are supported and all other options
> are ignored.

But based on the code changes, it seems like `WithInfo` is being used in places where `event.WriteError` is called. 

So it isn't clear to me:
- What the difference between `WithInfo` and `WithDetails` is.
- Why an Observation can take a `Detail`, but not `Info`.
- Why an Error can take `Info` but not `Detail`.

Documenting this on the `WriteError` and `WriteObservation` would be fine. Also if we do go with specific types instead of `map[string]interface` it might be a good spot to document the difference between a `Detail` and an `Info`.

Reply:
It looks like you made all of those changes while I was typing that response.

Reply:
Are you all set here then?

Reply:
Yep